gemini/sample-apps/fashion-trends-prediction/app/data_prep/driver.py

```diff
@@ -0,0 +1,331 @@
+                        influencer,
+                        [],
+                        scrape_size["num_posts"],
+                        cookies,
+                        model="Gemini",
+                    )
+
+                temp_map[influencer] = posts
+
+            time.sleep(1)
+
+            # save data after running for each influencer
+            with open(data_path, "w") as outfile:
+                json.dump(temp_map, outfile)
+
+    # delete the influencers that are not in the list of any country anymore
+    saved["global"] = {k: v for k, v in temp_map.items() if k in scraped_influencers}
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def periodic_extraction(scrape_size, first_period):
+    """Extracts data from Instagram periodically and saves it in a JSON file.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+            first_period (bool): A boolean indicating whether this is the first period.
+
+    """
+
+    if first_period:
+        saved = {}
+        saved["global"] = {}
+    else:
+        with open(data_path, "r") as f:
+            saved = json.load(f)
+
+    # first scrape is set to True when you are running for the first time in an extraction period
+    extract_json_influencer_wise(
+        scrape_size=scrape_size, first_scrape=True, saved=saved
+    )
+
+
+def create_final_data(scrape_size):
+    """Creates a final data file by combining data from Instagram and Vogue.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+
+    """
+
+    with open(data_path, "r") as f:
+        saved = json.load(f)
+
+    saved["finaldata"] = {}
+    for country_name in country_names[: scrape_size["num_countries"]]:
+        print(country_name)
+        saved["finaldata"][country_name] = {}
+
+        # the top influencers of that country right now
+        influencers = getInfluencers(country_name)
+
+        for influencer in influencers[: scrape_size["num_influencers"]]:
+            print(influencer)
+            if influencer in saved["global"]:
+                for element in saved["global"][influencer]:
+                    image_attr = element[2]
+
+                    for cat in image_attr:
+                        if cat in saved["finaldata"][country_name]:
+                            saved["finaldata"][country_name][cat].extend(
+                                image_attr[cat].copy()
+                            )
+                        else:
+                            saved["finaldata"][country_name][cat] = image_attr[
+                                cat
+                            ].copy()
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def merge_keys():
+    """Merges keys in the final data file that have only a difference of spaces, capital letters, etc."""
```

Comment:
I'm not sure if this function is needed. Since you're creating your own dictionaries for these, couldn't you use the same keys when defining them?

Reply:
The keys are returned by the LLM model. Sometimes, the model will return 'shirt' and sometimes 'Shirt'. Since, both of these are the same, we merge their contents.